storybookjs/storybook
# Feature/multiple hierarchies

Issue: #2451 

## What I did

Extended the hierarchy functionality to enable multiple hierarchies next to the main (nameless hierarchy).
I added an extra rootSeparator (not sure about the name though):
```
setOptions({
  // ... 
  hierarchySeparator: /\./,
  rootSeparator: /\|/, // default is false
});
```

**UPDATED**: `rootSeparator` is renamed to `hierarchyRootSeparator` with default value `null`

Extending the API as such:
```
storiesOf('Addons|Addon Info.React Docgen', module)
  .add(          ^
  ...
```

Works backwards compatible with or without setting the hierarchySeparator.

Documentation is updated in addons/options/README.md.

Allowing multiple hierarchies, see issue #2451 for screenshot.

## How to test

I added the rootSeparator to the kitchen sink example and added the "|" separator to the addon stories. 

See cra-kitchen-sink for an example.

The tests are fixed and extended. Though 1 test is failing, but that it is failing on master as well:
```
FAIL  examples/cra-kitchen-sink/src/storyshots.test.js
  ● Test suite failed to run

    TypeError: (0 , _react3.storiesOf)(...).add(...).add(...).add(...).addWithInfo is not a function
```
